README.md

```diff
@@ -1 +1,105 @@
+
+IPVM intends to do nothing less than connect all of the world's users and services. It can be thought of as "the HTTP of compute": open, interoperable, and everywhere. Following the model of the Web being the network of linked documents, IPVM is the network of linked computation. It allows anyone to permissionlessly tie into the network without prenegotiation.
+
+TCP/IP and HTTP were successful for a number of reasons. Networking was always possible, but doing this from scratch for every application is infeasible. Hypertext provided a single mechanism for many applications to interconnect. Application agnosticism is a major pillar of [such an approach][W3F Principles]: anything can run on this substrate, and it's easy to join. IPVM works equally well with traditional Cloud-based services as it does local-first compute and trustless architectures. To be successful, IPVM must have a path towards becoming substantially better than a pure Cloud architecture. It does not stand in opposition to the Cloud, but rather extends and contains it.
+
+The internet is no longer purely client/server. Consumer devices are now significantly more powerful than when the LAMP stack and Cloud infrastructure were being co-developed. Devices today are heterogeneous: there are powerful browsers, heavy servers, microservices, edge PoPs, tiny IoT devices, smartphones, and commons networks like BitTorrent, IPFS, and blockchains. IPVM provides a way to tie applications together with robust invocation, routing, and trust layers. The volume of data that modern applications requires leads to a situation known as data gravity. Given the speed of light, the amount of data that we produce and modify grows faster than is possible to sync to all locations, and .
+
+# Audience
+
+IPVM is intended for general distributed computation, but excels at tasks that would be found in systems like Lambda Step functions. The IPVM network itself is focused on machine-to-machine interaction — where each node acts as the user's agent — but is not expected to be interacted with directly when on the happy path.
+
+# Goals
+
+IPVM means to provide:
+
+- Task matchmaking network / marketplace
+- Pluggable execution engine
+- A trust layer (e.g. verifiable compute, encryption, PKI, capabilities)
+- Seamlessly tie together external services
+- Interfaces open to extension
+- First-class P2P payments
+
+IPVM is application agnostic, but intends to abstract concerns of locality away from the programmer.
+
+## Antigoals
+
+IPVM will _not_ attempt to develop into a:
+
+- replacement or upgrade of IPFS internals
+- high level language for distributed applications
+- custom Wasm runtime
+- blockchain or other global consensus mechanism
+
+IPVM's focus is on execution, networking, data flow, and trust. It is important to make the internals debuggable and friendly where possible, but only when it doesn't conflict with the core aim at this layer: efficiency and abstracting away distributed systems concerns.
```

Comment:
minor: I think this may read right as data flow (not one word), but we use dataflow later (one word).